src/librustc_mir/interpret/place.rs

```diff
@@ -451,6 +429,32 @@ where
         base.offset(offset, meta, field_layout, self)
     }
 
+    /// Index into an array.
+    #[inline(always)]
+    pub fn mplace_index(
+        &self,
+        base: MPlaceTy<'tcx, M::PointerTag>,
+        index: u64,
+    ) -> InterpResult<'tcx, MPlaceTy<'tcx, M::PointerTag>> {
```

Comment:
This is partially a duplication of logic in `mplace_field` -- but note that that already was a duplication of logic in the layout code. We cannot use the layout code as it is written for `usize`.